Sources/NineSunsSDK/NSUtils.swift

```diff
@@ -0,0 +1,37 @@
+//
+//  NSUtils.swift
+//  NineSunsSDK
+//
+//
+
+import Foundation
+import CryptoKit
+
+public struct NSUtils {
+  public static func hashEmail(_ email: String) -> String? {
+    guard let emailData = email.data(using: .utf8) else {
+      print("Failed to convert email to Data.")
+      return nil
+    }
+    let sha512Hash = SHA512.hash(data: emailData)
+    let hashData = Data(sha512Hash)
+    let base64HashString = hashData.base64EncodedString()
+    
+    guard let base64HashData = base64HashString.data(using: .utf8) else {
+      print("Failed to convert Base64 hash string to Data.")
+      return nil
+    }
+    let doubleBase64HashEmail = base64HashData.base64EncodedString()
+    return doubleBase64HashEmail
+  }
+  
```

Comment:
I think this function will be changed to `base64Encode` instead. Since hashing an email includes 2 actions hash (use the `NSUtils.hash512`) then use this `NSUtils.base64Encode` to encode the hash string under base64 format.

Reply:
i am confused on the base64encoding, what are we using that for?